topology/src/main/resources/proto/topology.proto

```diff
@@ -23,10 +24,29 @@ message PartitionState {
   int32 priority = 2;
 }
 
+message ClusterChangePlan {
+  int32 version = 1;
+  repeated TopologyChangeOperation operation = 2;
+}
+
+message TopologyChangeOperation {
+  string memberId = 1;
+  OperationType type = 2;
+  optional int32 partitionId = 3;
+  optional int32 priority = 4;
+}
```

Comment:
:thought_balloon: I think we could model this with `oneof`: https://protobuf.dev/programming-guides/proto3/#using-oneof

So something like 
```proto
message TopologyChangeOperation {
  string memberId = 1;
  oneof operation {
    PartitionJoinOperation partitionJoin = 2;
    PartitionLeaveOperation partitionLeave = 3;
  }
}
```

That would match our internal types and would make it easier to see which fields are needed for which operation.

Reply:
> For defining TopologyChangeOperation, partitionId and prioirity are added as optional fields. Alternative is to define a type for each of the operation type and use OneOf. But I think it is better to keep it simple, as it is just two optional fields. Besides it is unclear to me how [backward compatibility works with OneOf](https://protobuf.dev/programming-guides/proto3/#backward).

It is not clear to me if backwards compatibility will be an issue with OneOf. Could you take a look at the doc?

Reply:
I don't see anything concerning that would prevent us from making use of `oneof` here in the backwards compatibility docs: https://protobuf.dev/programming-guides/proto3/#backward